handler/oauth2/flow_authorize_code_auth.go

```diff
@@ -110,10 +110,15 @@ func (c *AuthorizeExplicitGrantHandler) IssueAuthorizeCode(ctx context.Context,
 	if err := c.CoreStorage.CreateAuthorizeCodeSession(ctx, signature, ar.Sanitize(c.GetSanitationWhiteList())); err != nil {
 		return errors.WithStack(fosite.ErrServerError.WithCause(err).WithDebug(err.Error()))
 	}
-
-	resp.AddQuery("code", code)
-	resp.AddQuery("state", ar.GetState())
-	resp.AddQuery("scope", strings.Join(ar.GetGrantedScopes(), " "))
+	if ar.GetRequestForm().Get("response_mode") == "form_post" {
```

Comment:
It looks like the two if statements are copies of one another. Can we avoid duplication here?

Reply:
yes it can be improved